docker/main/ngen/Dockerfile

```diff
@@ -333,30 +339,30 @@ RUN echo "export HYDRA_HOST_FILE=${HYDRA_HOST_FILE}" >> /etc/profile \
     && cd netcdf-fortran \
     && export NCDIR=/usr NFDIR=/usr \
     && LD_LIBRARY_PATH=/usr/lib CPPFLAGS=-I/usr/include LDFLAGS=-L/usr/lib ./configure --prefix=/usr \
-    && make -j $(nproc) && make install \
+    && make -j ${BUILD_PARALLEL_JOBS} && make install \
     ##### Build and install NetCDF C++ \
     && cd /tmp/ngen-deps \
     && mkdir netcdf-cxx4 \
     && tar -xzf netcdf-cxx4-${NETCDF_CXX_VERSION}.tar.gz -C netcdf-cxx4 --strip 1 \
     && mkdir netcdf-cxx4/build \
     && cd netcdf-cxx4/build \
     && cmake .. \
-    && make \
+    && make -j ${BUILD_PARALLEL_JOBS} \
     # TODO: if we run into any problem, might need to reactivate this \
     # && ctest \
     && make install \
     # Install required python dependency packages with Pip \
     && pip3 install numpy pandas pyyaml bmipy Cython blosc2 netCDF4 wheel packaging \
-    && HDF5_DIR=/usr pip3 install --install-option="\'--hdf5 \${HDF5_DIR}\'" --install-option="\'--jobs=$(nproc) \'"  --no-build-isolation tables \
+    && HDF5_DIR=/usr pip3 install --install-option="\'--hdf5 \${HDF5_DIR}\'" --install-option="\'--jobs=${BUILD_PARALLEL_JOBS} \'"  --no-build-isolation tables \
     # Make aliases for convenience \
     && alias pip='pip3' \
     && echo "alias pip='pip3'" >> /etc/profile \
     && alias python='python3'  \
     && echo "alias python='python3'" >> /etc/profile \
     # Also set up boost here, since we copied the download but only just installed bzip2 to work with it \
     && cd ${BOOST_ROOT} \
-    && tar -xjf boost_${BOOST_VERSION//./_}.tar.bz2 \
-    && rm boost_${BOOST_VERSION//./_}.tar.bz2 \
+    && tar -xf boost_tarball.blob \
+    && rm boost_tarball.blob \
```

Comment:
This may end up being irrelevant, depending on what's done about an earlier comment ...

I'm curious as to why this file name change was done.  I don't see any issues with it beyond a few uncommon and essentially trivial side effects when debugging image stages.  But the old way didn't exactly seem troublesome.  

Also, are we certain the `-j` flag isn't needed?